### PR TITLE
Refactored Brave Ads internal/base to internal/common

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/27088

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by aseren — APPROVED** [review]

**Review by kylehickinson — APPROVED** [review]
iOS++ if CI is good

File: chromium_presubmit_config.json5 [hunk 1]
```diff
@@ -126,11 +125,28 @@
...
+      "vendor/bat-native-ads/src/bat/ads/internal/ads/ad_events/ad_events_database_table\\.h",
+      "vendor/bat-native-ads/src/bat/ads/internal/ads/ad_events/search_result_ads/search_result_ad_event_handler\\.h",
+      "vendor/bat-native-ads/src/bat/ads/internal/conversions/conversion_queue_database_table\\.h",
+      "vendor/bat-native-ads/src/bat/ads/internal/conversions/conversions_database_table\\.h",
+      "vendor/bat-native-ads/src/bat/ads/internal/creatives/creative_ads_database_table\\.h",
+      "vendor/bat-native-ads/src/bat/ads/internal/creatives/inline_content_ads/creative_inline_content_ads_database_table\\.h",
+      "vendor/bat-native-ads/src/bat/ads/internal/creatives/new_tab_page_ads/creative_new_tab_page_ads_database_table\\.h",
+      "vendor/bat-native-ads/src/bat/ads/internal/creatives/promoted_content_ads/creative_promoted_content_ads_database_table\\.h",
+      "vendor/bat-native-ads/src/bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events\\.h",
+      "vendor/bat-native-ads/src/bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events_database_table\\.h",
```
Comment (tmancey) [inline chromium_presubmit_config.json5:149]: Will be fixed by https://github.com/brave/brave-browser/issues/26589

**Comment by tmancey** [comment]
The file `vendor/bat-native-ads/data/test/resources/feibnmjhecfbjpeciancnchbmlobenjn` has been shrunk to be smaller than 20mb to pass the presubmit checks.

**Review by iefremov — APPROVED** [review]

**Comment by iefremov** [comment]
chromium_src LGTM

**Pull request merged**